test/DynamoCoreTests/Graph/Nodes/NodeModelTests.cs

```diff
@@ -54,20 +58,68 @@ public void DictionaryLinkTest()
         [Category("UnitTests")]
         public void SelectNeighborsTest()
         {
-            string listTestFolder  = Path.Combine(TestDirectory, "core");
+            //Open and run the workspace
+            string listTestFolder = Path.Combine(TestDirectory, "core");
             string testFilePath = Path.Combine(listTestFolder, "Angle.dyn");
-            
+
             RunModel(testFilePath);
 
+            //Select the node with neighbors
             NodeModel node = CurrentDynamoModel.CurrentWorkspace.NodeFromWorkspace("dcd9c6c6-6350-4292-a553-c57a764504b4");
-            
+
             var countBefore = DynamoSelection.Instance.Selection.Count;
             Assert.AreEqual(0, countBefore);
 
+            //Run the method and assert whether more nodes were selected
             node.SelectNeighbors();
 
             var countAfter = DynamoSelection.Instance.Selection.Count;
             Assert.AreEqual(2, countAfter);
         }
+
+        /// <summary>
+        /// Class created in order to test protected methods in the NodeModel parent
+        /// </summary>
+        private class NodeModelTestingClass : NodeModel
+        {
+            public bool UpdateValueCoreBool(UpdateValueParams uvParams) => 
+                UpdateValueCore(uvParams);
+
+            public NodeMigrationData MigrateToDsFunctionNoAssembly(NodeMigrationData nmData, string name, string funcName) => 
+                MigrateToDsFunction(nmData, name, funcName);
+
+            public NodeMigrationData MigrateToDsFunctionAssembly(NodeMigrationData nmData, string assembly, string name, string funcName) => 
+                MigrateToDsFunction(nmData, assembly, name, funcName);
+
+            public NodeMigrationData MigrateToDsVarArgFunctionNMData(NodeMigrationData data, string assembly, string name, string funcName) => 
```

Comment:
Any tests for the other three functions here?

Reply:
I am currrently working on those tests. Those methods will be used there.

Reply:
@byorda-glb Cool, thanks!